src/topology/compact_open.lean

```diff
@@ -138,10 +141,46 @@ lemma nhds_compact_open_eq_Inf_nhds_induced (f : C(α, β)) :
 by { rw [compact_open_eq_Inf_induced], simp [nhds_infi, nhds_induced] }
 
 lemma tendsto_compact_open_iff_forall {ι : Type*} {l : filter ι} (F : ι → C(α, β)) (f : C(α, β)) :
-  filter.tendsto F l (nhds f)
+  filter.tendsto F l (𝓝 f)
   ↔ ∀ s (hs : is_compact s), filter.tendsto (λ i, (F i).restrict s) l (𝓝 (f.restrict s)) :=
 by { rw [compact_open_eq_Inf_induced], simp [nhds_infi, nhds_induced, filter.tendsto_comap_iff] }
 
+/-- A family `F` of functions in `C(α, β)` converges in the compact-open topology, if and only if
+it converges in the compact-open topology on each compact subset of `α`. -/
+lemma exists_tendsto_compact_open_iff_forall [locally_compact_space α] [t2_space α] [t2_space β]
+  {ι : Type*} {l : filter ι} [filter.ne_bot l] (F : ι → C(α, β)) :
+  (∃ f, filter.tendsto F l (𝓝 f))
+  ↔ ∀ (s : set α) (hs : is_compact s), ∃ f, filter.tendsto (λ i, (F i).restrict s) l (𝓝 f) :=
```

Comment:
Would it be useful to the end-user if there were companion lemmas that say "those `f`s for each `s` are the restriction of `f` to each `s`" and in the other direction "and all these `f`s can be glued together, obtaining a function to which `F` converges"?

No problem if you think either "it wouldn't be that useful" or "it can happen later"!

Reply:
I've added a couple of more general lemmas which imply this information: (1) if a family `F` converges to `f`, then the restrictions of `F` to `s` converge to the restriction of `f` to `s`; (2) under appropriate hypotheses `continuous_map` is a Hausdorff space (hence, limits are unique).  I think these are useful so thanks for prompting this!

Reply:
I'd be tempted to have the explicit lemmas in both directions, to be sure we really have the API available, but I can see the argument that the explicit lemmas might not often be needed. I'll hit `bors d+` now, but do feel free to add more if you like! :-)

Reply:
Oh!  My instinct is the opposite, that the gluing construction `lift_cover_restrict'` is an implementation detail for which it's better to expose only certain carefully selected properties :-). I'll take your `bors d+` for now, but I expect that when @CBirkbeck and others start using this construction heavily, we'll find out what's best.